File: iterator/java/implementations/Library.java
package iterator.java.implementations;

import iterator.java.interfaces.Book;
import iterator.java.interfaces.Iterator;

// implementations/Library.java
public class Library {
    public static void displayBooks(Iterator iterator) {
        for (Book book = iterator.first(); !iterator.isDone(); iterator.next(), book = iterator.currentItem()) {
            System.out.println("Libro: " + book.getTitle() + " - Autor: " + book.getAuthor());
        }
    }
}

File: iterator/python/implementations/library.py
from interfaces.iterator import Iterator

class Library:
    @staticmethod
    def display_books(iterator: Iterator) -> None:
        book = iterator.first()
        while not iterator.is_done():
            print(f"Libro: {book.get_title()} - Autor: {book.get_author()}")
            iterator.next()
            book = iterator.current_item()